sensors/nxt_i2c_sensor_core.c

```diff
@@ -377,6 +399,18 @@ static int nxt_i2c_sensor_detect(struct i2c_client *client,
 #endif
 
 	/*
+	 * NXT Temperature sensor detection is handled here.
+	 */
+	if (client->addr == 0x4c) {
+		ret = nxt_i2c_sensor_detect_lego_temp(client);
+		if (ret)
```

Comment:
Hmm... If there are ever any other sensors with this address, I thing we would want this to be...

    if (ret == 0) {
        snprintf(info->type, I2C_NAME_SIZE, "%s",
                 LEGO_NXT_TEMPERATURE_SENSOR_NAME);
        return 0;
    }

And then let flow fall though to the regular detection if it is not a temperature sensor.